docs: mention filtered by ACLs in coordinate API (#12489)
hashicorp/consul: website/content/partials/http_api_results_filtered_by_acls.mdx

> The HTTP response header `X-Consul-Results-Filtered-By-ACLs: true` will be present
> if some results were filtered out due to ACL policy configuration.
> Refer to the [HTTP API documentation](/api-docs#results-filtered-by-acls) for more information.

Reviewer: ```suggestion
The `X-Consul-Results-Filtered-By-ACLs: true` HTTP header appears in the response if a subset of the results have been filtered out due to ACL policy configuration.
Refer to the [HTTP API documentation](/api-docs#results-filtered-by-acls) for more information.
```
I think we want to use the simple present tense as much as possible. The past perfect is good for describing why something turned out the way it did.

Author: Or:
```
The HTTP response includes the `X-Consul-Results-Filtered-By-ACLs: true` header
if the response array excludes any results due to ACL policy configuration.
Refer to the [HTTP API documentation](/api-docs#results-filtered-by-acls) for more information.
```

Author: I think I prefer the second alternative I gave because it explains it differently than the header name, giving the user two ways of explaining / understanding the same thing. "Results filtered by ACLs" and "response array excludes results".

I think having the header name say "results filtered by ACLs" and then saying "array filtered by ACLs" is just restating the header name without making it any easier to understand.

Reviewer: ```
The HTTP response includes the `X-Consul-Results-Filtered-By-ACLs: true` header
if the response array excludes results that as a result of an ACL policy configuration.
Refer to the [HTTP API documentation](/api-docs#results-filtered-by-acls) for more information.
```
Really starting to split hairs here, but I agree with the idea that the response is an _array_ and that some of the results were _excluded_ as a result of an ACL policy configuration linked with the token.

Author: ### On "any"
I agree that the word "any" is extraneous, so I'll remove it.

### On "due to" versus "as a result of"
I still think
> excludes results due to ACL policy configuration

is better than
> excludes results as a result of an ACL policy configuration

because we avoid repetition of the word "result".

### Next steps

If I remove the word "any", is this ready to be marked approved? I can't actually merge because you still have an outstanding "changes requested" review.
